api/v1/storageprofile_types.go

```diff
@@ -24,25 +24,25 @@ import (
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
 // StorageProfileSpec defines the desired state of StorageProfile
-// +kubebuilder:validation:XValidation:rule="!has(oldSelf.deviceClass) || has(self.deviceClass)", message="deviceClass is required once set"
-// +kubebuilder:validation:XValidation:rule="!has(oldSelf.sharedFilesystemConfiguration) || has(self.sharedFilesystemConfiguration)", message="sharedFilesystemConfiguration is required once set"
-// +kubebuilder:validation:XValidation:rule="!has(oldSelf.blockPoolConfiguration) || has(self.blockPoolConfiguration)", message="blockPoolConfiguration is required once set"
+// +kubebuilder:validation:XValidation:rule="has(oldSelf.deviceClass) && has(self.deviceClass)", message="deviceClass is required"
+// +kubebuilder:validation:XValidation:rule="has(oldSelf.sharedFilesystemConfiguration) && has(self.sharedFilesystemConfiguration)", message="sharedFilesystemConfiguration is required"
+// +kubebuilder:validation:XValidation:rule="has(oldSelf.blockPoolConfiguration) && has(self.blockPoolConfiguration)", message="blockPoolConfiguration is required"
 type StorageProfileSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="DeviceClass is immutable"
 	// +kubebuilder:validation:MaxLength=512
 	// DeviceClass is the deviceclass name.
 	DeviceClass string `json:"deviceClass"`
 
-	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="SharedFilesystemConfiguration is immutable"
 	// configurations to use for cephfilesystem.
 	SharedFilesystemConfiguration SharedFilesystemConfigurationSpec `json:"sharedFilesystemConfiguration,omitempty"`
 
-	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Required
```

Comment:
AFAIK this is optional not a mandatory to have it

Reply:
See above

Reply:
https://github.com/red-hat-storage/ocs-operator/pull/2231#discussion_r1371772674